databases/week2/HW/school-query.sql

```diff
@@ -0,0 +1,33 @@
+-- Part 2: School database
+-- Create a new database containing the following tables:
+-- Class: with the columns: id, name, begins (date), ends (date)
+-- Student: with the columns: id, name, email, phone, class_id (foreign key)
+-- If you are done with the above tasks, you can continue with these advanced tasks:
+-- Create an index on the name column of the student table.
+-- Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations).
+
+CREATE TABLE `class` (
+  `id` int unsigned NOT NULL AUTO_INCREMENT,
+  `name` varchar(255) NOT NULL,
+  `begins` DATE NOT NULL,
+  `ends` DATE NOT NULL,
+  PRIMARY KEY (`id`)
+);
+
+CREATE TABLE `student` (
+  `id` int unsigned NOT NULL AUTO_INCREMENT,
+  `name` varchar(255) NOT NULL,
+  `email` varchar(255) NOT NULL,
+  `phone` varchar(255) NULL,
+  `class_id` int(10) unsigned,
+  PRIMARY KEY (`id`),
+  FOREIGN KEY (`class_id`) REFERENCES `class` (`id`)
+);
+
+-- Create an index on the name column of the student table.
+CREATE INDEX `student_name` 
+ON `student` (name);
+
+-- Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations).
+ALTER TABLE class
+ADD status ENUM('not-started', 'ongoing', 'finished') NOT NULL;
```

Comment:
In this case one problem willl occur if the student has more than one classes.
to solve this we need a student_classes table which can bind a student to more classes.

regarding this query
```

ALTER TABLE class
ADD status ENUM('not-started', 'ongoing', 'finished') NOT NULL

```
a default value must be added to the query because you already define it as NOT NULL.
When NOT NULL the already existing rows will have problem because they can not have an empty value in the status column.

Reply:
Than you Babak for the detailed feedback. I understood now only few suggestions made.I have made the changes as mentioned.

Reply:
ALTER TABLE class
ADD status ENUM('not-started', 'ongoing', 'finished') NOT NULL DEFAULT 'not started'

i forgot to add this query in the changes i made